internal/provider/resource_access_secret.go

```diff
@@ -377,3 +389,59 @@ func (s *accessSecretResource) trace(msg string, additionalFields ...map[string]
 	}
 	tflog.SubsystemTrace(s.subCtx, LogResourceAccessSecret, msg, additionalFields...)
 }
+
+// UpgradeState upgrades the state of the offer resource.
+// This is used to handle changes in the resource schema between versions.
+func (o *accessSecretResource) UpgradeState(ctx context.Context) map[int64]resource.StateUpgrader {
+	return map[int64]resource.StateUpgrader{
+		// Upgrade from `endpoint` to `endpoints` attribute.
```

Comment:
Upgrade from applications ListAttribute to SetAttribute?

Reply:
done